codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java

```diff
@@ -430,6 +430,22 @@ private static MqttFixedHeader decodeFixedHeader(ByteBuf buffer) {
         return new Result<String>(s, result.numberOfBytesConsumed);
     }
 
+    private static Result<byte[]> decodeBytes(ByteBuf buffer, int minBytes, int maxBytes) {
+        final Result<Integer> decodedSize = decodeMsbLsb(buffer);
+        int size = decodedSize.value;
+        int numberOfBytesConsumed = decodedSize.numberOfBytesConsumed;
+        if (size < minBytes || size > maxBytes) {
+            buffer.skipBytes(size);
+            numberOfBytesConsumed += size;
+            return new Result<byte[]>(null, numberOfBytesConsumed);
+        }
+        byte[] b = new byte[size];
+        buffer.getBytes(buffer.readerIndex(), b);
```

Comment:
replace the call to `getBytes` and `skipBytes` with a call to `buffer.readBytes(b);`